### PR TITLE
envoy: 1.27.5 -> 1.30.1

### LINKED ISSUE #287027: Update request: envoy 1.27.2 → 1.29.0
- Package name: envoy
- Latest released version: v1.29.0

- Current version on the unstable channel: v1.27.2
- Current version on the stable/release channel: v1.27.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@lukegb 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by malt3**
## Description of changes

Fixes https://github.com/NixOS/nixpkgs/issues/287027

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/http/envoy/default.nix [hunk 1]
```diff
@@ -172,6 +168,14 @@ buildBazelPackage {
     "--java_runtime_version=local_jdk"
     "--tool_java_runtime_version=local_jdk"
 
+    # undefined reference to 'grpc_core::*Metadata*::*Memento*
+    #
+    # During linking of the final binary, we see undefined references to grpc_core related symbols.
+    # The missing symbols would be instantiations of a template class from https://github.com/grpc/grpc/blob/v1.59.4/src/core/lib/transport/metadata_batch.h
+    # "ParseMemento" and "MementoToValue" are only implemented for some types
+    # and appear unused and unimplemented for the undefined cases reported by the linker.
+    "--linkopt=-Wl,--unresolved-symbols=ignore-in-object-files"
```
Comment (malt3) [inline pkgs/servers/http/envoy/default.nix:177]: Here is the output containing all undefined symbols. If someone understands why this happens, I would be happy to apply a cleaner fix.

https://gist.github.com/malt3/37745a0c97966f09b997be201770d7be

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 306529` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>envoy</li>
    <li>pomerium</li>
  </ul>
</details>

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 306529` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>envoy</li>
    <li>pomerium</li>
  </ul>
</details>

**Review by lukegb — APPROVED** [review]
Nice! I was working on getting this done as well, but thanks for beating me to it (and finding some of the rough edges here...)

**Comment by adamcstephens** [comment]
How would we feel about supporting multiple envoy versions? :smiling_face_with_tear:

**Comment by malt3** [comment]
> How would we feel about supporting multiple envoy versions? 🥲

I personally have nothing against that, except that the full 1.27 line will be EOL soon.

Maybe we could keep the latest version supported by consul and the actually latest version of envoy?

**Comment by adamcstephens** [comment]
1.28.x is the latest currently supported by consul. https://developer.hashicorp.com/consul/docs/connect/proxies/envoy#supported-versions

**Comment by malt3** [comment]
I can give that a try if @lukegb as the official maintainer agrees.

**Comment by lukegb** [comment]
If you're going to add 1.28.x, please do not add it until after the NixOS 24.05 branch off; it will go EOL before NixOS 24.05 goes EOL. 1.29.x would be fine but that's not super helpful for Envoy...

**Comment by malt3** [comment]
@adamcstephens What I don’t quite understand is if this support matrix is actually an upper bound in practice.
Do you know that consul actually breaks with newer versions of envoy?
Since I am not using consul, it is hard for me to understand how much of a hard requirement this really is.

**Comment by adamcstephens** [comment]
Consul can be configured to ignore the version restriction, which means it's possibly ok unless something has been removed that is relied on. I'll test this PR on one of my nodes and see if 1.30.1 is ok before we dive into maintaining multiple versions.

**Comment by adamcstephens** [comment]
1.30.1 looks like it works fine with consul connect 1.18.1. Nevermind then. :)

**Pull request merged**